backend/layers/processing/h5ad_data_file.py

```diff
@@ -95,19 +98,23 @@ def to_cxg(
 
     def write_anndata_x_matrices_to_cxg(self, output_cxg_directory, ctx, sparse_threshold):
         matrix_container = f"{output_cxg_directory}/X"
-
         x_matrix_data = self.anndata.X
-        is_sparse = is_matrix_sparse(x_matrix_data, sparse_threshold)  # big memory usage
-        logging.info(f"is_sparse: {is_sparse}")
-
-        convert_matrices_to_cxg_arrays(matrix_container, x_matrix_data, is_sparse, ctx)  # big memory usage
+        with dask.config.set(
+            {
+                "num_workers": 2,  # match the number of workers to the number of vCPUs
```

Comment:
Use the default value for `num_works`, `threads_per_worker`, and `distributed.worker.memory.limit`use. The default values are calculated based on the available resources. it will be easier to maintain since we might decide to change the resource allocation to the job in TF.

Reply:
I tried to at first, but kept running into misc. memory issues on some datasets. Most common was oversplitting and not having enough memory allocated on a thread to load the chunk it was processing. Did you ever run into issues like that in your testing?